### ISSUE TITLE
Update VideoJS library

### ISSUE THREAD

**Issue by Teranode** [pydio/pydio-core#869]
I think it would benefit some if the videojs player were to be updated to the latest, I have done it already, but I think it would be good as an official update.

**Comment by cdujeu** [comment]
Hi Christian, was it anything complicated? Can you maybe submit a PR with your changes? Did you just drop in the latest code of videojs ?

**Comment by Teranode** [comment]
i don't necessarily know how github fully works so here is a pastebin of the file i edited:
http://pastebin.com/cBTRQbZD

Then go to: /plugins/editor.video/video-js/
and add all of the latest videojs files in there (including the lang and font and swf)

you will no longer need "player_swf_maxi.swf" so that can be deleted.

And then its simple updates for users, they would just have to upload a new version of videojs (if they want to) to the video-js fold and then clear cache and then it will be updated.

IF they need to change to a custom videojs, they have to go into class.VideoPreviewer.js and change "vjs-default-skin". (Cant really explain this portion cause all that is explained in videojs's docs.)

**Comment by cdujeu** [comment]
would be really cool to learn how to submit a PR... :-) But well, i'll see when I have some time
-c

**Comment by cdujeu** [comment]
last but not least: can you summarize the advantages of upgrading?

**Comment by Teranode** [comment]
Most of my time is spent coding and i dont like using github a lot (or at all)

Ultimately the benefits of upgrading would be to utilize videojs's functions towards being flexible so lets say for instance you want to use the "ads" plugin videojs has a s a contribution, you can utilize that with pydio (if you know the right coding). I necessarily did it for my own benefit because i like to keep up to date with basically anything. And using videojs 2.0.2 (or whatever version 2 it was on)
wasnt going to cut it. So it would be nice to have that upgrade "in" pydio instad of having to change it  "every" time. (as mentioned before you can easily just upload the files and replace and it will be updated). Plus it has a nice mobile/html5 compatability.

**Comment by Teranode** [comment]
I'm also going to take a look into optimizing the VideoReader class. RIght now it uses PHP to gather metadata about the file, and it's semi-resource intensive when loading.

An interesting thing to use would be XSendFile which is more optimal when using php for file data reading.

**Comment by cdujeu** [comment]
Christian we do already provide an option for using xsendfile, this happens un access.fs and Is probably already used by the reader class.
That said, if you want to carry on contributing, you will please go through the standard process (signing CLA and submitting proper github PR), thank you in advance :)

**Comment by cdujeu** [comment]
Droping your version of viewer and the newest library, i have resizing issue: select a file in the list, the video display in the list panel (correct size), but if I resize the panel, the video does not resize correctly (see screenshot). Same for you?
![capture d ecran 2015-03-19 a 15 52 39](https://cloud.githubusercontent.com/assets/2465057/6733096/3808d08a-ce50-11e4-9acf-e4f75cb4c5aa.png)

**Comment by Teranode** [comment]
Here is an updated version of it: 
http://pastebin.com/EyYWcbef

**Comment by cdujeu** [comment]
Still not working correctly for me, resizing is random, JS error on div 'vjs-flash-playback', and opening in full screen (using "Open" action) is giving black screen. 
Please test toroughly on many browsers before submitting again!
-c

**Comment by cdujeu** [comment]
Hi @Teranode any updated version working well everywhere?

**Comment by cdujeu** [comment]
Updated in develop
